File: CodingSamples/Java/Runtime/ReflectionTest2/src/PersonalLoan.java
package finance;

public class PersonalLoan {

	@MaxDuration(6)
	public float common(double amount, int period) {
		return 10 + 0.5f * period;
	}

	@MaxDuration(7)
	public float employee(double amount, int period) {
		return common(amount, period) / 2;
	}
}

File: CodingSamples/DotNet/Runtime/ReflectionTest2/FinLib/PersonalLoan.cs
namespace Finance;

public class PersonalLoan
{
    [MaxDuration(Limit = 6)]
    public float Common(double amount, int period) => 10 + 0.5f * period;

    public float Employee(double amount, int period) => amount < 600000 ? Common(amount, period) / 2 : 8;
}
